src/components/common/IDContainer.test.ts

```diff
@@ -0,0 +1,41 @@
+import { renderHook } from '@testing-library/react-hooks'
+
+import { useUniqueIdFn } from './IdContainer'
+
+
+type LodashExports = typeof import('lodash/fp')
+jest.mock('lodash/fp', (): LodashExports => {
+  const actual = jest.requireActual<LodashExports>('lodash/fp')
+
+  let uniqueSeed = 123
+
+  return ({
+    ...actual,
+    uniqueId: () => {
+      const result = uniqueSeed
+      uniqueSeed++
+      return result.toString(10)
+    }
+  })
+})
+
+describe('useUniqueIdFn', () => {
+  it('returns function to use for unique Id', () => {
+    // Arrange
+    const getMyId = renderHook(useUniqueIdFn).result.current
+    const getMyId2 = renderHook(useUniqueIdFn).result.current
+
+    // Act
+    const buttonAId = getMyId('button-a')
+    const buttonBId = getMyId('button-b')
+    const otherId = getMyId2('other')
+
+    // Assert
+    // id can be reused within a component and will be same seed
```

Comment:
Why would you ever want to reuse the id? I think of the point of this function just that it gives you an id that is guaranteed to be unique on the page.

Reply:
Right now the code is spitting out non-human-readable ids.  Everything is prefixed with 'element-123uniqueToken123'.  The intention was to make things a bit more human readable.  The ID would be intended to be re-used, and sub-components would tag on different prefixes... hence the examples of button-a-123, button-b-123. 

The proposed implementation has the hook return a callable function, which just prefixes a cached unique ID for the component which survives multiple render cycles.  This would have the benefit of also helping debugging since the ID tokens would match for a given component's sub-components.

But as was stated in other comments, this should be an uncommon need that we can largely remove with tweaks/additions to our common component offerings and/or recommended patterns for things like form labels, tab controls, etc.

I do see the benefit of @nawatts suggestion to just have the hook return the raw id.  That makes the useage more transparent....

Reply:
also, the intent was to primarily protect against duplicated instances of a control within the application... so that instanceA of MyCoolComponent and instanceB elsewhere in the app can have unique Ids from eachother.   The specific proposed solution is to solve this by also making the Ids more human-readable and debug-friendly.

Reply:
_"The ID would be intended to be re-used, and sub-components would tag on different prefixes... hence the examples of button-a-123, button-b-123."_

I am really not convinced this was the original intention, and it has the potential of folks creating non-unique IDs, which defeats the purpose of having the utility method. There is really no reason that an ID needs to be human-readable-- it is purely used to associate elements together, and as long as you are assigning it to a variable, you don't care what it is.

Why would you want to encourage the `button-a-123` pattern?

Reply:
I still disagree that the IDs in their current form cause pain (or at least, I haven't heard of pain… is it documented somewhere?). As long as the ID is short, it is simple to search for it in the DOM. And dev tools do a nice job of linking things together. I think the risk of people creating non-unique IDs outweighs the potential benefit of a human-readable ID.

![image](https://user-images.githubusercontent.com/484484/208507399-da1f15a6-d41b-4201-91ed-d0eecd9addcd.png)

Reply:
even the original IDContainer solution has intentional durability of the generated id between render cycles to avoid headaches of the id value mutating more often then expected (in between render cycles).  But the comments here have given me some great feedback in general.

I think I will simplify to @nawatts suggestion of having the hook simply return a durable unique ID, and have an optional argument at hook creation to allow for a prefix, which will default to 'element' and have a hypen added internally... matching #element-85 example in screen shot above.